scripts/yosys/logic_equiv_check.tcl

```diff
@@ -35,18 +36,19 @@ if { [info exists ::env(VERILOG_FILES_BLACKBOX)] } {
 	}
 }
 foreach lib $::env(LIB_SYNTH_COMPLETE_NO_PG) {
-	read_liberty -nooverwrite -lib -ignore_miss_dir -setattr blackbox $lib
```

Comment:
You're replacing the liberty file in the `foreach` loop with a different environment variable.

Reply:
Because the liberty file provided by $lib environment is liberty file with no pg connections and you can only read it as black-box but there is no meaning of reading it as a black-box because when liberty file is read as black-box then during lec yosys can't figure out the logic of cells used in netlist and always fails the lec.

Reply:
Then you update `foreach lib $::env(LIB_SYNTH_COMPLETE_NO_PG) {` to `foreach lib $::env(LIB_SYNTH) {`. As it stands in your PR the for loop is meaningless and there is an inconsistency issue where you're looping over `LIB_SYNTH_COMPLETE_NO_PG` and then reading `LIB_SYNTH`.

Reply:
Updated